### ISSUE TITLE
cache_configuration cannot be set to Null (None in Python) to remove AFD Route Caching once turned on

### ISSUE THREAD

**Issue by jartoo** [Azure/azure-sdk-for-python#35801]
- **Package Name**: azure-mgmt-cdn
- **Package Version**: 13.1.0
- **Operating System**: MacOS M1 ARM
- **Python Version**: 3.10.11

**Describe the bug**
When using the RouteUpdateParameters class to remove Caching from an existing Route, it requires you to pass null, which in Python uses the None keyword, to the cache_configuration object.  It does not remove the caching when trying to update the Route.

**To Reproduce**
1. Create an Azure Front Door Standard CDN Profile
2. Create an Endpoint, Origin-Group, Origin and Rule Set.
3. Create a Route with Caching enabled (you can select any options as the options do not matter
4. Attempt to update the Route to remove the Caching, by calling RouteOperations.begin_update(), which requires you to set the cache_configuration to "null", which uses the None keyword
5. Check the caching status in the portal to confirm it was disabled and you will see it is not.

**Expected behavior**
The Caching should then be disabled, since there is no explicit variable to toggle the caching on or off, like exists in the Azure Portal.

**Screenshots**
N/A

**Additional context**
Tracing the Python SDK seems to remove all the variables set to None as part of the _serialization.py when executing the _routes_operations.py begin_update function.  Since the cache_configuration object is set to None I cannot figure out how to get it to pass a Null to the API.  Am I missing something here?  Thanks.

**Comment by ChenxiJiang333** [comment]
Hi @jartoo, python sdk is just a tool to call the rest apis and is generated by swagger definitions. The related definition is [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-02-01/afdx.json#L5567), and it seems there's no parameter that can help disable the caching, so the sdk can't do it either. 
The only way according to the definition that to disable the caching is to not enable it when creating the route though api or sdk. Or, to request such feature, you can open an issue at https://github.com/Azure/azure-rest-api-specs/issues, once the swagger is updated, python will release a new version to support it as well.

**Comment by msyyc** [comment]
Hi @jartoo, it is by design to filter `None` when serialization. And you has a special type `NULL` of `azure.core` if you really want to pass `null` in request payload. Here is example: https://github.com/Azure-Samples/azure-samples-python-management/blob/040589b4a529a228a096e2c6605284fbe7eea495/samples/resources/manage_resource_group_tag.py#L11

**Comment by jartoo** [comment]
Let me test this, thanks!  I missed the NULL type...

**Comment by jartoo** [comment]
It seems when I call the routes.begin_create() using the AzureCoreNull (NULL type per your example) in the azure-mgmt-cdn==13.1.0 module, the _serialization.py key extractor function attribute_key_case_insensitive_extractor() seems to throw an exception when passing it this type.  It appears to have three key extractors when I trace the code.  This is the second one that throws the exception.

To reproduce:
```python
from azure.identity import DefaultAzureCredential
from azure.mgmt.cdn import CdnManagementClient
from azure.mgmt.cdn.models import Route, RouteUpdateParameters, CompressionSettings, ResourceReference, AfdRouteCacheConfiguration
from azure.core.serialization import NULL as AzureCoreNull

client = CdnManagementClient(credential=DefaultAzureCredential(), subscription_id='xxxxxxx')

cache_configuration = AzureCoreNull
parameters = Route(
    cache_configuration=cache_configuration
)
client.routes.begin_create(
    resource_group_name="myResourceGroup",
    profile_name="myFrontDoor",
    endpoint_name="myEndPoint",
    route_name="myRoute",
    route=parameters
)
```
Result:
File ".venv/lib/python3.11/site-packages/azure/mgmt/cdn/_serialization.py", line 706, in body
    raise SerializationError("Unable to build a model: " + str(err)) from err
azure.core.exceptions.SerializationError: Unable to build a model: Unable to deserialize to object: type

Maybe I am missing something?

**Comment by msyyc** [comment]
Let me take a look.

**Comment by msyyc** [comment]
Hi @jartoo Please try again with latest package: https://pypi.org/project/azure-mgmt-cdn/13.1.1/

**Comment by jartoo** [comment]
Will work on this early next week and cycle back with feedback, thanks!!

**Comment by jartoo** [comment]
Looks like this issue is resolved in 13.1.1 when testing with the Rules() object.  Thanks!

I will cycle back if I see this persist in any other objects.  Agree we can close this at this time.
